packages/compiler/src/collector/expression.ts

```diff
@@ -1,49 +1,124 @@
 import { Node, ts } from "ts-morph";
 import { match } from "ts-pattern";
+import * as types from "../types";
 
-export const handleJsxExpression = (node: Node<ts.Node>) => {
+/**
+ * Normalize ts-morph Node to make further processing simpler
+ */
+export const normalizeNode = (node: Node): Node => {
+  if (Node.isAsExpression(node)) {
+    return normalizeNode(node.getExpression());
+  }
+
+  if (Node.isParenthesizedExpression(node)) {
+    return normalizeNode(node.getExpression());
+  }
+
+  if (Node.isNonNullExpression(node)) {
+    return normalizeNode(node.getExpression());
+  }
+
+  if (Node.isTypeAssertion(node)) {
+    return normalizeNode(node.getExpression());
+  }
+
+  return node;
+};
+
+export const handleJsxExpression = (
+  node: Node<ts.Node>
+): types.Value | undefined => {
```

Comment:
Note: Now the semantics of the return value of this function is clearer: `undefined` means "failed to evaluate" and `types.Value` can contain `undefined`. Previously these are not separated and `<Box color={undefined} />` was not statically compiled since `undefined` means "it failed to extract in the logic"